### ISSUE TITLE
The details pane does not trap keyboard focus when using Shift + Tab

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#4376]
## Describe the bug

[Test URL]
https://clarity.design/documentation/datagrid/detail-pane

[WCAG 2.1 Guideline]
Success Criterion 2.4.3 Focus Order
Understanding Focus Order|
(Level A)
https://www.w3.org/TR/WCAG21/#focus-order

[OS]
macOS, Windows

[User Agents/Assistive Technologies]
macOS Chrome (keyboard-only), Windows Firefox (keyboard-only) macOS VoiceOver/Safari

[Component]
Datagrid Detail Pane

[Description]
The details pane does not trap keyboard focus when using Shift + Tab. 

See the relevant HTML/CSS below.

[User Impact]
Keyboard-only and screen reader users may be confused when using the 
Shift + Tab key combination to move browser focus backwards through the browser focus order and browser focus moves to elements outside of the details pane. 

[Recommendation]
Trap browser focus within the Datagrid Detail Pane when the user is pressing Shift + Tab 

## How to reproduce

[Steps to Reproduce]
- Go the the Test URL
- Activate one of the Datagrid Detail Panes and observe that browser focus moves to it
- Repeatedly press the Tab key and observe that browser focus is trapped within the Datagrid Detail Pane
- Press Shift + Tab repeatedly and observe that browser focus is not trapped within the Datagrid Detail Pane as expected
- Repeat the steps above with listed browser and screen reader combinations and observe browser focus is not trapped within the Datagrid Detail Pane as expected and the screen is silent as elements outside of the Datagrid Detail Pane gain browser focus.

## Expected behavior
Browser focus should be trapped within the Datagrid Detail Pane when the user is pressing Shift + Tab

**Comment by gnomeontherun** [comment]
This is a generic problem with our focus trap, it doesn't support this anywhere. Screen readers can hear the boundaries as a way to detect the end. 

Closing as a duplicate of https://github.com/vmware/clarity/issues/2400.
